bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/AmcrestHandler.java

```diff
@@ -0,0 +1,248 @@
+
+    public AmcrestHandler(ThingHandler handler) {
+        ipCameraHandler = (IpCameraHandler) handler;
+    }
+
+    public void setURL(String url) {
+        requestUrl = url;
+    }
+
+    // This handles the incoming http replies back from the camera.
+    @Override
+    public void channelRead(@Nullable ChannelHandlerContext ctx, @Nullable Object msg) throws Exception {
+        if (msg == null || ctx == null) {
+            return;
+        }
+        try {
+            String content = msg.toString();
+
+            if (!content.isEmpty()) {
+                ipCameraHandler.logger.trace("HTTP Result back from camera is \t:{}:", content);
+            }
+            if (content.contains("Error: No Events")) {
+                if ("/cgi-bin/eventManager.cgi?action=getEventIndexes&code=VideoMotion".equals(requestUrl)) {
+                    ipCameraHandler.noMotionDetected(CHANNEL_MOTION_ALARM);
+                } else if ("/cgi-bin/eventManager.cgi?action=getEventIndexes&code=AudioMutation".equals(requestUrl)) {
+                    ipCameraHandler.noAudioDetected();
+                }
+            } else if (content.contains("channels[0]=0")) {
+                if ("/cgi-bin/eventManager.cgi?action=getEventIndexes&code=VideoMotion".equals(requestUrl)) {
+                    ipCameraHandler.motionDetected(CHANNEL_MOTION_ALARM);
+                } else if ("/cgi-bin/eventManager.cgi?action=getEventIndexes&code=AudioMutation".equals(requestUrl)) {
+                    ipCameraHandler.audioDetected();
+                }
+            }
+
+            if (content.contains("table.MotionDetect[0].Enable=false")) {
+                ipCameraHandler.setChannelState(CHANNEL_ENABLE_MOTION_ALARM, OnOffType.valueOf("OFF"));
+            } else if (content.contains("table.MotionDetect[0].Enable=true")) {
+                ipCameraHandler.setChannelState(CHANNEL_ENABLE_MOTION_ALARM, OnOffType.valueOf("ON"));
+            }
+            // determine if the audio alarm is turned on or off.
+            if (content.contains("table.AudioDetect[0].MutationDetect=true")) {
+                ipCameraHandler.setChannelState(CHANNEL_ENABLE_AUDIO_ALARM, OnOffType.valueOf("ON"));
+            } else if (content.contains("table.AudioDetect[0].MutationDetect=false")) {
+                ipCameraHandler.setChannelState(CHANNEL_ENABLE_AUDIO_ALARM, OnOffType.valueOf("OFF"));
+            }
+            // Handle AudioMutationThreshold alarm
+            if (content.contains("table.AudioDetect[0].MutationThreold=")) {
+                String value = ipCameraHandler.returnValueFromString(content, "table.AudioDetect[0].MutationThreold=");
+                ipCameraHandler.setChannelState(CHANNEL_THRESHOLD_AUDIO_ALARM, PercentType.valueOf(value));
+            }
+
+        } finally {
+            ReferenceCountUtil.release(msg);
+            ctx.close();
+        }
+    }
+
+    public String encodeSpecialChars(String text) {
+        String Processed = text;
```

Comment:
Please use camelcase for variable names.
```suggestion
        String processed = text;
```